streams/src/main/java/org/apache/kafka/streams/kstream/internals/foreignkeyjoin/CombinedKey.java

```diff
@@ -14,26 +14,25 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
 package org.apache.kafka.streams.kstream.internals.foreignkeyjoin;
 
 import java.util.Objects;
 
-public class CombinedKey<KF, KP> {
-    private final KF foreignKey;
-    private final KP primaryKey;
+public class CombinedKey<KRight, KLeft> {
```

Comment:
It's a bit weird that right is left and left is right. But not sure if I have a solution as we cannot swap them. How about standardizing on `KForeign` and `KPrimary`? That seemed rather clear to me, but I may be lacking context.

Reply:
The order is on-purpose, because that is the actually order of both arguments of `CombinedKey`, ie, especially how we serialize it. I was also first thinking about maybe swapping it, but in the end, I did keep as, as it might be better to align the order to the serialization order?

Let me think about the names a little bit more, so it can sink it :)